plugins/plugin-testing-java/plugin-testing-classpath/src/main/java/org/eclipse/che/plugin/java/testing/AbstractJavaTestRunner.java

```diff
@@ -0,0 +1,143 @@
+            LOG.debug("Can't read all methods.", e);
+        }
+
+        return result;
+    }
+
+    private void addAllTestsMethod(List<TestPosition> result, ICompilationUnit compilationUnit) throws JavaModelException {
+        for (IType type : compilationUnit.getAllTypes()) {
+            for (IMethod method : type.getMethods()) {
+                if (isTestMethod(method, compilationUnit)) {
+                    result.add(createTestPosition(method));
+                }
+            }
+        }
+    }
+
+    private TestPosition createTestPosition(IMethod method) throws JavaModelException {
+        ISourceRange nameRange = method.getNameRange();
+        ISourceRange sourceRange = method.getSourceRange();
+
+        return DtoFactory.newDto(TestPosition.class).withFrameworkName(getName())
+                         .withTestName(method.getElementName())
+                         .withTestNameStartOffset(nameRange.getOffset())
+                         .withTestNameLength(nameRange.getLength())
+                         .withTestBodyLength(sourceRange.getLength());
+    }
+
+    protected abstract boolean isTestMethod(IMethod method, ICompilationUnit compilationUnit);
+
+    protected IJavaProject getJavaProject(String projectPath) {
+        IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectPath);
+        return JavaModelManager.getJavaModelManager().getJavaModel().getJavaProject(project);
+    }
+
+    protected ICompilationUnit findCompilationUnitByPath(IJavaProject javaProject, String filePath) {
+        try {
+            IClasspathEntry[] resolvedClasspath = javaProject.getResolvedClasspath(false);
+            IPath packageRootPath = null;
+            for (IClasspathEntry classpathEntry : resolvedClasspath) {
+                if (filePath.startsWith(classpathEntry.getPath().toOSString())) {
+                    packageRootPath = classpathEntry.getPath();
+                    break;
+                }
+            }
+
+            if (packageRootPath == null) {
+                throw getRuntimeException(filePath);
+            }
+
+            String packagePath = packageRootPath.toOSString();
+            if (!packagePath.endsWith("/")) {
+                packagePath += "/";
+            }
+
+            String pathToClass = filePath.substring(packagePath.length());
+            IJavaElement element = javaProject.findElement(new Path(pathToClass));
+            if (element != null && element.getElementType() == IJavaElement.COMPILATION_UNIT) {
+                return (ICompilationUnit)element;
+            } else {
+                throw getRuntimeException(filePath);
+            }
+        } catch (JavaModelException e) {
+            throw new RuntimeException("Can't find Compilation Unit.", e);
+        }
+    }
+
+    @Override
+    public int getDebugPort() {
+        return -1;
+    }
+
+    private RuntimeException getRuntimeException(String filePath) {
```

Comment:
Method returns RuntimeException when have a problem to find java element by path (package or compilation unit). It is an inner private method.